cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tether/TetherStore.java

```diff
@@ -0,0 +1,215 @@
+      StructuredTable tetherTable = context
+        .getTable(StoreDefinition.TetherStore.TETHER);
+      CloseableIterator<StructuredRow> iterator = tetherTable
+        .scan(Range.all(), Integer.MAX_VALUE);
+      iterator.forEachRemaining(row -> {
+        String peerName = row.getString(StoreDefinition.TetherStore.PEER_NAME_FIELD);
+        String endpoint = row.getString(StoreDefinition.TetherStore.PEER_URI_FIELD);
+        TetherStatus tetherStatus = TetherStatus.valueOf(row.getString(StoreDefinition.TetherStore.TETHER_STATE_FIELD));
+        PeerMetadata peerMetadata = GSON.fromJson(row.getString(StoreDefinition.TetherStore.PEER_METADATA_FIELD),
+                                                  PeerMetadata.class);
+        long lastConnectionTime = row.getLong(StoreDefinition.TetherStore.LAST_CONNECTION_TIME_FIELD);
+        peers.add(new PeerInfo(peerName, endpoint, tetherStatus, peerMetadata, lastConnectionTime));
+      });
+      return peers;
+    });
+  }
+
+  /**
+   * Get information about a peer
+   *
+   * @return information about status of a peer
+   * @throws IOException if reading from the database fails
+   * @throws PeerNotFoundException if the peer is not found
+   */
+  public PeerInfo getPeer(String peerName) throws IOException, PeerNotFoundException {
+    return TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable tetherTable = context
+        .getTable(StoreDefinition.TetherStore.TETHER);
+      Range range =  Range.singleton(
+        ImmutableList.of(Fields.stringField(StoreDefinition.TetherStore.PEER_NAME_FIELD, peerName)));
+      try (CloseableIterator<StructuredRow> iterator = tetherTable.scan(range, Integer.MAX_VALUE)) {
+        if (!iterator.hasNext()) {
+          throw new PeerNotFoundException(peerName);
+        }
+        StructuredRow row = iterator.next();
+        String endpoint = row.getString(StoreDefinition.TetherStore.PEER_URI_FIELD);
+        TetherStatus tetherStatus = TetherStatus.valueOf(row.getString(StoreDefinition.TetherStore.TETHER_STATE_FIELD));
+        PeerMetadata peerMetadata = GSON.fromJson(row.getString(StoreDefinition.TetherStore.PEER_METADATA_FIELD),
+                                                  PeerMetadata.class);
+        long lastConnectionTime = row.getLong(StoreDefinition.TetherStore.LAST_CONNECTION_TIME_FIELD);
+        return new PeerInfo(peerName, endpoint, tetherStatus, peerMetadata, lastConnectionTime);
+
+      }
+    });
+  }
+
+  /**
+   * Get tether status for a peer
+   *
+   * @param peerName name of the peer
+   * @return tether status
+   * @throws IOException if reading from the database fails
+   * @throws PeerNotFoundException if the peer is not found
+   */
+  public TetherStatus getTetherStatus(String peerName) {
+    return TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable tetherTable = context
+        .getTable(StoreDefinition.TetherStore.TETHER);
+      Range range =  Range.singleton(
```

Comment:
Why need to scan? The `peerName` is the primary key, right?

Reply:
changed to read()